### ISSUE TITLE
format_args!() inlining/flattening allows for longer lifetimes

### ISSUE THREAD

**Issue by m-ou-se** [rust-lang/rust#110769]
Before #109999, this gave an error:

```rust
let f = format_args!("{}", "abc"); // error[E0716]: temporary value dropped while borrowed
dbg!(f);
```

But after #109999, that no longer gives an error.

Ideally it shouldn't give an error in any case, but that's [a separate issue](https://github.com/rust-lang/rust/issues/92698). Inlining/flattening format_args shouldn't change what compiles and what doesn't compile, so, for now, this still shouldn't compile.

**Comment by m-ou-se** [comment]
Luckily the fix is easy. We just need to make sure that any `format_args!(…)` with arguments still results in an expansion that involves borrowing a temporary.
